lib/embeddings/semantic_search.rb

```diff
@@ -43,28 +44,42 @@ def search_for_topics(query, page = 1)
           DiscourseAi::Embeddings::VectorRepresentations::Base.current_representation(strategy)
 
         digest = OpenSSL::Digest::SHA1.hexdigest(search_term)
-        hyde_key = build_hyde_key(digest, SiteSetting.ai_embeddings_semantic_search_hyde_model)
 
-        embedding_key =
-          build_embedding_key(
-            digest,
-            SiteSetting.ai_embeddings_semantic_search_hyde_model,
-            SiteSetting.ai_embeddings_model,
-          )
-
-        hypothetical_post =
-          Discourse
-            .cache
-            .fetch(hyde_key, expires_in: 1.week) { hypothetical_post_from(search_term) }
-
-        hypothetical_post_embedding =
-          Discourse
-            .cache
-            .fetch(embedding_key, expires_in: 1.week) { vector_rep.vector_from(hypothetical_post) }
+        search_embedding = nil
+
+        if hyde
+          hyde_key = build_hyde_key(digest, SiteSetting.ai_embeddings_semantic_search_hyde_model)
+
+          embedding_key =
+            build_embedding_key(
+              digest,
+              SiteSetting.ai_embeddings_semantic_search_hyde_model,
+              SiteSetting.ai_embeddings_model,
+            )
+
+          hypothetical_post =
+            Discourse
+              .cache
+              .fetch(hyde_key, expires_in: 1.week) { hypothetical_post_from(search_term) }
+
+          search_embedding =
+            Discourse
+              .cache
+              .fetch(embedding_key, expires_in: 1.week) do
+                vector_rep.vector_from(hypothetical_post)
+              end
+        else
+          embedding_key = build_embedding_key(digest, "", SiteSetting.ai_embeddings_model)
+
+          search_embedding =
+            Discourse
+              .cache
+              .fetch(embedding_key, expires_in: 1.week) { vector_rep.vector_from(search_term) }
+        end
```

Comment:
I think that this method is having too much responsibility. What would you say about extracting 2 private methods like `search_embedding_with_hyde` and `search_embedding_without_hyde`. 

Then:
```ruby
search_embedding = if hyde
  search_embedding_with_hyde
else
  search_embedding_without_hyde
end
```